File: backend/src/main/java/com/example/backend/model/OrderDetail.java
package com.example.backend.model;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class OrderDetail {
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;
        private int amount;
        private int price;
        @ManyToOne (fetch = FetchType.LAZY)
        @JoinColumn(name="order_id")
        private Order order;

        public OrderDetail(Long id) {
                this.id = id;
        }

}

File: frontend/src/app/entity/OrderDetail.ts
import { Order } from "./Order";
import { Product } from "./Product";

export class OrderDetail {

  id: number | null = null;
  product: Product | null = null;
  amount: number | null = null;
  price: number | null = null;
  order: Order | null = null;

  constructor(o?: Partial<OrderDetail>) {
    Object.assign(this, o);
  }
}
